src/views/AssignPickerModal.vue

```diff
@@ -71,14 +80,153 @@ export default defineComponent({
     IonTitle,
     IonToolbar,
   },
+  computed: {
+    ...mapGetters({
+      openOrders: 'order/getOpenOrders',
+      currentFacility: 'user/getCurrentFacility'
+    })
+  },
+  data () {
+    return {
+      selectedPickers: [],
+      queryString: '',
+      pickers: []
+    }
+  },
   methods: {
+    isPickerSelected(id) {
+      return this.selectedPickers.some((picker) => picker.id == id)
+    },
     closeModal() {
       modalController.dismiss({ dismissed: true });
     },
+    pickerChanged(id) {
+      const picker = this.selectedPickers.some((picker) => picker.id == id)
+      if (picker) {
+        // if picker is already selected then removing that picker from the list on click
+        this.selectedPickers = this.selectedPickers.filter((picker) => picker.id != id)
+      } else {
+        this.selectedPickers.push(this.pickers.find((picker) => picker.id == id))
+      }
+    },
+    async searchPicker () {
+      this.pickers = []
+      this.fetchPickers()
+    },
+    async printPicklist () {
+      if (!this.selectedPickers.length) {
+        showToast(translate('Select a picker'))
+        return;
+      }
+
+      emitter.emit("presentLoader")
+      let resp;
+
+      const orders = this.state.order.open.list;
+      const items = []
+
+      const formData = new FormData();
+      formData.append("facilityId", this.currentFacility.facilityId);
+      orders.map((order) => {
+        order.doclist.docs.map((item) => items.push(item))
+      });
+
+      items.map((item, index) => {
+        formData.append("facilityId_o_"+index, this.currentFacility.facilityId)
+        formData.append("shipmentMethodTypeId_o_"+index, item.shipmentMethodTypeId)
+        formData.append("itemStatusId_o_"+index, "PICKITEM_PENDING")
+        formData.append("shipGroupSeqId_o_"+index, item.shipGroupSeqId)
+        formData.append("orderId_o_"+index, item.orderId)
+        formData.append("orderItemSeqId_o_"+index, item.orderItemSeqId)
+        formData.append("productId_o_"+index, item.productId)
+        formData.append("quantity_o_"+index, item.itemQuantity)
+        formData.append("inventoryItemId_o_"+index, item.inventoryItemId)
+        formData.append("picked_o_"+index, item.itemQuantity)
+        formData.append("_rowSubmit_o_"+index, "Y")
+      })
```

Comment:
We could skip an iteration
```suggestion
     int index = 0;
      orders.map((order) => {
        order.doclist.docs.map((item) => {
          formData.append("facilityId_o_"+index, this.currentFacility.facilityId)
        formData.append("shipmentMethodTypeId_o_"+index, item.shipmentMethodTypeId)
        formData.append("itemStatusId_o_"+index, "PICKITEM_PENDING")
        formData.append("shipGroupSeqId_o_"+index, item.shipGroupSeqId)
        formData.append("orderId_o_"+index, item.orderId)
        formData.append("orderItemSeqId_o_"+index, item.orderItemSeqId)
        formData.append("productId_o_"+index, item.productId)
        formData.append("quantity_o_"+index, item.itemQuantity)
        formData.append("inventoryItemId_o_"+index, item.inventoryItemId)
        formData.append("picked_o_"+index, item.itemQuantity)
        formData.append("_rowSubmit_o_"+index, "Y")
        index++;
        } 
      });
```